pyqtorch/primitive.py

```diff
@@ -228,11 +266,45 @@ def __init__(self, gate: str, control: int | tuple[int, ...], target: int):
             n_control_qubits=len(self.control),
         ).squeeze(2)
         super().__init__(mat, target)
+        self.gate = globals()[gate]
         self.qubit_support = self.control + (self.target,)  # type: ignore[operator]
 
     def extra_repr(self) -> str:
         return f"control:{self.control}, target:{(self.target,)}"
 
+    def tensor(
+        self, values: dict[str, Tensor] = {}, n_qubits: int = 1, diagonal: bool = False
+    ) -> Tensor:
+        from pyqtorch.circuit import Sequence
+
+        if diagonal:
+            raise NotImplementedError
+        if n_qubits < max(self.qubit_support) + 1:
+            n_qubits = max(self.qubit_support) + 1
+        proj1 = Sequence(
+            [Projector(qubit_support=qubit, ket="1", bra="1") for qubit in self.control]
+        )
+        """
+        proj0 = Sequence(
+            [Projector(qubit_support=qubit, ket="0", bra="0") for qubit in self.control]
+        )
+        Don't work for multiple control qubit (want to keep it ?)
+        c_mat = proj0.tensor(values, n_qubits) + operator_product(
+            proj1.tensor(values, n_qubits),
+            self.gate(self.target).tensor(values, n_qubits),
+            self.target,
+        )
+        """
+        c_mat = (
+            I(target=self.control[0]).tensor(values, n_qubits)
+            - proj1.tensor(values, n_qubits)
+            + operator_product(
+                proj1.tensor(values, n_qubits),
+                self.gate(self.target).tensor(values, n_qubits),
+            )
+        )
+        return c_mat
```

Comment:
Related to the previous comments, using the `Sequence` here feels a bit too hacky...